### PR TITLE
Implement refactored notifications

### PR DESCRIPTION
**Pull request by Abyss777**
Web part for https://github.com/traccar/traccar/pull/3950

image

`tagfield` automatically set comma-separated values

image

Also improved `arrayListFilter` to handle comma-separated strings

image

Some comments in code...

### CODE REVIEW

File: web/app/Application.js [hunk 1]
```diff
@@ -128,6 +129,11 @@ Ext.define('Traccar.Application', {
         return Strings[key] || key;
     },
 
+    getNotificatorString: function (eventType) {
+        var key = 'notification' + eventType.charAt(0).toUpperCase() + eventType.slice(1);
```
Comment (Abyss777) [inline web/app/Application.js]: It is better use here `'notificator'` but we already have translated strings

File: web/app/AttributeFormatter.js [hunk 2]
```diff
@@ -73,7 +73,16 @@ Ext.define('Traccar.AttributeFormatter', {
     },
 
     deviceIdFormatter: function (value) {
-        return Ext.getStore('Devices').getById(value).get('name');
+        var device, store;
+        if (value !== 0) {
```
Comment (Abyss777) [inline web/app/AttributeFormatter.js]: Made it consistent with `xxxIdFormatter` mostly to handle if `deviceId = 0`

File: web/app/controller/Root.js [hunk 3]
```diff
@@ -289,6 +290,8 @@ Ext.define('Traccar.controller.Root', {
                     this.beep();
                 }
                 Traccar.app.showToast(array[i].text, device.get('name'));
+            } else {
+                Traccar.app.showToast(array[i].text);
```
Comment (Abyss777) [inline web/app/controller/Root.js]: Handle a test event without `deviceId`

File: web/l10n/en.json [hunk 4]
```diff
@@ -359,6 +359,7 @@
     "notificationWeb": "Send via Web",
     "notificationMail": "Send via Mail",
     "notificationSms": "Send via SMS",
+    "notificationNotificators": "Way of Notification",
```
Comment (tananaev) [inline web/l10n/en.json]: I think we should just name this "Channels".

File: web/l10n/en.json [hunk 5]
```diff
@@ -359,6 +359,7 @@
     "notificationWeb": "Send via Web",
```
Comment (tananaev) [inline web/l10n/en.json]: What do you think about renaming these to just "web", "email" and "SMS"?

File: web/app/view/ArrayListFilter.js [hunk 6]
```diff
@@ -34,6 +34,13 @@ Ext.define('Traccar.view.ArrayListFilter', {
                         return true;
                     }
                 }
+            } else if (property.indexOf(',') !== -1) {
```
Comment (tananaev) [inline web/app/view/ArrayListFilter.js]: What if it's space?

File: web/app/store/AllNotificators.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+Ext.define('Traccar.store.AllNotificators', {
+    extend: 'Ext.data.Store',
+    model: 'Traccar.model.KnownNotificator',
+
+    proxy: {
+        type: 'rest',
+        url: 'api/notifications/notificators',
+        listeners: {
+            'exception': function (proxy, response) {
```
Comment (tananaev) [inline web/app/store/AllNotificators.js]: Do we need quotes around `exception`?

File: web/app/view/ArrayListFilter.js (hunk 6 above)
Comment (Abyss777) [reply web/app/view/ArrayListFilter.js]: Backend does not handle such case (space-separated).
But I can implement it in this custom component if we plan to use it somewhere else.

File: web/l10n/en.json (hunk 5 above)
Comment (Abyss777) [reply web/l10n/en.json]: OK. Then we can use `"notificator"` as prefix...

File: web/app/view/ArrayListFilter.js (hunk 6 above)
Comment (tananaev) [reply web/app/view/ArrayListFilter.js]: I think it should be consistent. Your code below handles spaces.

File: web/l10n/en.json (hunk 5 above)
Comment (tananaev) [reply web/l10n/en.json]: Sure.

**Comment by Abyss777** [comment]
All done. Removed quotation from a couple other places.

**Comment by tananaev** [comment]
Cool. I think it looks good. Can I merge both pull requests now?

**Comment by Abyss777** [comment]
Forget screenshot
image

Yes I believe both are ready now.

**Comment by tananaev** [comment]
Merged both. I'm thinking of implementing push notifications for Traccar Manager. It would be nice to have.

**Comment by Abyss777** [comment]
I'm very waiting this. I even installed this project https://pushjet.io/ and wanted to implement own 'notificator' for this.

Moreover you already on the dark side and used firebase in client :)

You can count on my help.

**Comment by tananaev** [comment]
I wouldn't call it a dark side. We still provide alternative without Google libraries.

**Comment by Abyss777** [comment]
Yes, sorry. It is really important to have alternative.

**Pull request merged**